src/bom/extract.rs

```diff
@@ -63,37 +65,17 @@ pub fn do_bitcode_extraction(extract_options : &ExtractOptions,
     // Thus the more robust solution is to specify the output file in a temporary
     // directory, and there is already a convenient temporary directory created
     // above to hold the output llvm bitcode tar file.
-    let mut objcopy_args = Vec::new();
-    objcopy_args.push(OsString::from("--dump-section"));
-    objcopy_args.push(OsString::from(format!("{}={}", ELF_SECTION_NAME, ok_tar_name)));
-
-    objcopy_args.push(OsString::from(&extract_options.input));
-
-    let mut objres = PathBuf::new();
-    objres.push(tmp_path);
-    objres.push("discard{output-file}");
-    objcopy_args.push(OsString::from(objres));
-
-    match Command::new("objcopy").args(&objcopy_args).spawn() {
-        Err(msg) => {
-            return Err(anyhow::Error::new(ExtractError::ErrorRunningCommand(String::from("objcopy"), objcopy_args, msg)));
-        }
-        Ok(mut child) => {
-            match child.wait() {
-                Err(msg) => {
-                    return Err(anyhow::Error::new(ExtractError::ErrorRunningCommand(String::from("objcopy"), Vec::new() /*objcopy_args*/, msg)));
-                }
-                Ok(sts) => {
-                    if !sts.success() {
-                        match sts.code() {
-                            Some(rc) => { return Ok(rc) }
-                            None => { return Ok(-1) }
-                        }
-                    }
-                }
-            }
-        }
-    }
+    let mut dummy_output = PathBuf::from(tmp_path);
+    dummy_output.push("discard{output-file}");  // arbitrary name
+
+    extract_ops.push_op(
+        &SubProcOperation::new(&Executable::new("objcopy",
+                                                ExeFileSpec::Append,
+                                                ExeFileSpec::Append))
+            .set_label("objcopy:extract-bitcode")
+            .push_arg("--dump-section")
+            .push_arg(format!("{}={}", ELF_SECTION_NAME, ok_tar_name))
+            .set_output_file(&FileArg::loc(dummy_output)));
```

Comment:
Would it make sense to have some kind of primitive in this library to just discard output? Doesn't have to go in this PR, but it seems like something that might come up.

Reply:
There is a primitive to discard output (the `NoFileUsed` constructor at https://kquick.github.io/chainsop/chainsop/enum.ExeFileSpec.html), but in this case, we explicitly need an output file because of: https://github.com/travitch/build-bom/blob/master/src/bom/extract.rs#L40-L65